### PR TITLE
Issue #7668 - Add tags for AWS glue resources

### PR DESCRIPTION
**Pull request by yozamacs**
This is a work in progress. I started with the AWS glue job resource first just to make sure that I had the necessary changes to add tags for one resource. As per https://aws.amazon.com/about-aws/whats-new/2019/03/aws-glue-now-supports-resource-tagging/ AWS glue supports tags on crawlers, endpoints, jobs, and triggers. I wasn't sure if there was other documentation that needed to be changed when you update a resource. Once I get a review of this, I can go ahead and implement tags on the other AWS glue resources 

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
AWS Glue resources now support tagging
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: aws/resource_aws_glue_job.go [hunk 1]
```diff
@@ -161,6 +162,10 @@ func resourceAwsGlueJobCreate(d *schema.ResourceData, meta interface{}) error {
 		input.SecurityConfiguration = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("tags"); ok {
```
Comment (yozamacs) [inline aws/resource_aws_glue_job.go]: It's unclear to me whether the tags need to be addressed in the read and update functions

File: aws/resource_aws_glue_job.go [hunk 2]
```diff
@@ -107,6 +107,7 @@ func resourceAwsGlueJob() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"tags": tagsSchema(),
```
Comment (yozamacs) [inline aws/resource_aws_glue_job.go]: I'm also not sure if this is the correct tag function or if it should be `tagsSchemaForceNew()`

**Comment by yozamacs** [comment]
@ztane I addressed that issue. Do you have any other feedback?

**Comment by bflad** [comment]
Hi @yozamacs 👋 Thanks for starting work on this. We recently simplified and documented the process for adding resource tagging to the Contributing Guide: https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#resource-tagging-code-implementation

The internal library already supports Glue tagging so you can skip the code generation instructions, but its important to follow the other instructions such as creating a new acceptance test and test configurations instead of adjusting the existing testing.

Please reach out with any questions or if you do not have time to implement this further.

File: aws/resource_aws_glue_job.go (hunk 2 above)
Comment (bflad) [reply aws/resource_aws_glue_job.go]: The resource does not need to be replaced to update its resource tags, so `tagsSchema()` is appropriate. 👍

File: aws/resource_aws_glue_job.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_glue_job.go]: Tags should be handled in Create, Read, and Update functions. See also: https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#resource-tagging-code-implementation

**Comment by bflad** [comment]
Hi again @yozamacs 👋 Since we didn't hear back from you about the above, we are going to close this for now. If you or anyone else is motivated to work on this following the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#resource-tagging-code-implementation), we will be happy to take a look again. 👍

**Pull request closed without merging**